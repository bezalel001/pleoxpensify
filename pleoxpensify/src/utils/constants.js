export const LIMIT = 25;
export const BASE_URL = 'http://localhost:3000/expenses';
