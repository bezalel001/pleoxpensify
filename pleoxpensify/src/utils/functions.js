import moment from 'moment';

export const formatAmount = ({ currency, value }) => {
  return new Intl.NumberFormat('en-UK', {
    style: 'currency',
    currency
  }).format(parseInt(value, 10));
};

export const formatDate = date => moment(date).format('MMM Do, YYYY');
