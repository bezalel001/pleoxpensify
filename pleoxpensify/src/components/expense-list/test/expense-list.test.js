import React from 'react';
import { shallow } from 'enzyme';
import ExpenseList from '../index';
import { expenses } from '../../../utils/fixtures';

describe('Expense List', () => {
  it('should render ExpenseList with expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render ExpenseList with empty message', () => {
    const wrapper = shallow(<ExpenseList expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should create an `li` per list item', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses} />);
    expect(wrapper.find('li').length).toBe(expenses.length);
  });
});
