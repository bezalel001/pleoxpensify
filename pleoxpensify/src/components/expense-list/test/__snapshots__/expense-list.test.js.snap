// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense List should render ExpenseList with emoty message 1`] = `ShallowWrapper {}`;

exports[`Expense List should render ExpenseList with empty message 1`] = `ShallowWrapper {}`;

exports[`Expense List should render ExpenseList with expenses 1`] = `ShallowWrapper {}`;
